### ISSUE TITLE
 Wrong key encoding throws error  `dataWritten == dataLength' failed

### ISSUE THREAD

**Issue by hitodama** [barrysteyn/node-scrypt#55]
```
var scrypt = require("scrypt");
var scryptParameters = scrypt.params(0.1);
scrypt.hash.config.keyEncoding = "base64"; /*wrong*/
scrypt.hash('foool', scryptParameters); /*<-- encoded ascii*/

node: ../src/node-boilerplate/common.cc:264: int Internal::ProduceBuffer(v8::Handle<v8::Value>&, const string&, std::string&, const node::encoding&, bool): Assertion `dataWritten == dataLength' failed.
Aborted

scrypt@3.0.1
```

**Comment by barrysteyn** [comment]
I will make it produce a more appropriate error

**Comment by barrysteyn** [comment]
Hi @hitodama 

You have a good point. There should not be any exceptions thrown. The problem is that changing the code may break it to the point where people will have to rewrite their libraries again. Perhaps we can have a compatibility mode? I am up for suggestions, what do you think?

**Comment by barrysteyn** [comment]
@hitodama Sorry for the obtuse answer above. You are absolutely correct, this is a bug. I will fix it ASAP.

**Comment by barrysteyn** [comment]
This is resolved in  commit d1b62e57b373f3b51ab77083738153a3bac1a115

**Comment by barrysteyn** [comment]
Note: I will roll out this fix when I roll out the Nan updated
